### ISSUE TITLE
Failed to list *v2alpha1.CronJob: the server could not find the requested resource

### ISSUE THREAD

**Issue by jicki** [kubernetes/kube-state-metrics#295]
```
kubectl get nodes
NAME          STATUS    ROLES     AGE       VERSION
172.16.1.64   Ready     <none>    20d       v1.8.1
172.16.1.65   Ready     <none>    20d       v1.8.1
172.16.1.66   Ready     <none>    20d       v1.8.1
```

```
ld not find the requested resource
E1106 07:09:40.573135       1 reflector.go:205] k8s.io/kube-state-metrics/collectors/cronjob.go:93: Failed to list *v2alpha1.CronJob: the server could not find the requested resource
E1106 07:09:41.574269       1 reflector.go:205] k8s.io/kube-state-metrics/collectors/cronjob.go:93: Failed to list *v2alpha1.CronJob: the server could not find the requested resource
E1106 07:09:42.575712       1 reflector.go:205] k8s.io/kube-state-metrics/collectors/cronjob.go:93: Failed to list *v2alpha1.CronJob: the server could not find the requested resource
E1106 07:09:43.576884       1 reflector.go:205] k8s.io/kube-state-metrics/collectors/cronjob.go:93: Failed to list *v2alpha1.CronJob: the server could not find the requested resource
E1106 07:09:44.578064       1 reflector.go:205] k8s.io/kube-state-metrics/collectors/cronjob.go:93: Failed to list *v2alpha1.CronJob: the server could not find the requested resource
E1106 07:09:45.581133       1 reflector.go:205] k8s.io/kube-state-metrics/collectors/cronjob.go:93: Failed to list *v2alpha1.CronJob: the server could not find the requested resource
E1106 07:09:46.582299       1 reflector.go:205] k8s.io/kube-state-metrics/collectors/cronjob.go:93: Failed to list *v2alpha1.CronJob: the server could not find the requested resource
E1106 07:09:47.583628       1 reflector.go:205] k8s.io/kube-state-metrics/collectors/cronjob.go:93: Failed to list *v2alpha1.CronJob: the server could not find the requested resource
E1106 07:09:48.584764       1 reflector.go:205] k8s.io/kube-state-metrics/collectors/cronjob.go:93: Failed to list *v2alpha1.CronJob: the server could not find the requested resource
E1106 07:09:49.586064       1 reflector.go:205] k8s.io/kube-state-metrics/collectors/cronjob.go:93: Failed to list *v2alpha1.CronJob: the server could not find the requested resource

```

pls  help me `

**Comment by zouyee** [comment]
running  `apiserver` with  --runtime-config=batch/v2alpha1=true

**Comment by drampelt** [comment]
I'm also having this issue, running kube-state-metrics 1.1.0 on Kubernetes 1.8 on GKE. CronJobs are now enabled for GKE but according to the [documentation](https://cloud.google.com/container-engine/docs/how-to/jobs) are under the `batch/v1` version, not `batch/v2alpha1`

**Comment by Morriz** [comment]
+1

any update on this? I am not able to run kubernetes 1.8 now which is out for a while. Why not make the version configurable?

**Comment by brancz** [comment]
Yes, we'll be updating the API used by kube-state-metrics for the next release. @andyxning @zouyee could one of you take care of this?

**Comment by sorenmat** [comment]
Does this mean that it doesn't ship metrics to datadog?

**Comment by sorenmat** [comment]
Yeah, that was a strange comment. What i meant was:
Does this break the metric collection or can the error just be ignored?

**Comment by brancz** [comment]
I see :slightly_smiling_face: this only affects the `CronJob` metrics, everything else continues to work as expected. Each collector has it's own go routine, and if one errors, the remaining ones keep doing their work.

**Comment by brancz** [comment]
tl;dr unless you want `CronJob` metrics, this error can be ignored. (you can also turn off the cronjob collector :slightly_smiling_face: )

**Comment by errordeveloper** [comment]
@brancz this can be closed now, a patch release would be very handy!

**Comment by brancz** [comment]
I don't think this should be shipped in a patch release. A minor release should be relatively close though. I would just like to get #302 in. Then I think we have enough actual features together to justify a minor.

**Comment by zouyee** [comment]
agree with @brancz

**Comment by denibertovic** [comment]
@brancz How does one turn off the cronjob collector?

**Comment by brancz** [comment]
You need to specify all collectors except the cronjob one in the —collectors flag.

**Comment by denibertovic** [comment]
@brancz You mean in `kube-state-metrics-deployment.yaml` at the bottom of the file as an argument to pod_nanny?

**Comment by denibertovic** [comment]
For some reason when I add this and do kubectl apply again it errors out with:
"a container name must be specified for pod kube-state-metrics-59fb6c5b94-pddz6"

**Comment by brancz** [comment]
this may have to do with kubectl apply, try kubectl replace

**Comment by denibertovic** [comment]
I tried that, I also tried deleting the service, pods and deployment first ... and then just applying everything again but no go.

**Comment by brancz** [comment]
When I use the exact manifest, it works for me. Can you share in more detail what you are trying to do, and the manifests you are using to accomplish this?

**Comment by denibertovic** [comment]
@brancz This is the change I made:
```git
--- a/kube-state-metrics/kubernetes/kube-state-metrics-deployment.yaml
+++ b/kube-state-metrics/kubernetes/kube-state-metrics-deployment.yaml
@@ -14,6 +14,20 @@ spec:
       containers:
       - name: kube-state-metrics
         image: quay.io/coreos/kube-state-metrics:v1.1.0
+        args:
+          - --collectors=daemonsets
+          - --collectors=deployments
+          - --collectors=limitranges
+          - --collectors=nodes
+          - --collectors=replicasets
+          - --collectors=replicationcontrollers
+          - --collectors=resourcequotas
+          - --collectors=services
+          - --collectors=jobs
+          - --collectors=statefulsets
+          - --collectors=persistentvolumeclaims
+          - --collectors=namespaces
+          - --collectors=horizontalpodautoscalers
```

**Comment by zouyee** [comment]
It like that `- --collectors=daemonsets, pods, ...`
